### ISSUE TITLE
provision vm from clone leads to panic interface {} is nil, not string

### ISSUE THREAD

**Issue by lethargosapatheia** [Telmate/terraform-provider-proxmox#814]
Hello,

I'm trying to provision a virtual machine from a template in proxmox (6.4-15) and I get this error:
```
╷
│ Error: Plugin did not respond
│
│   with proxmox_vm_qemu.proxmox_vm["vm_name"],
│   on main.tf line 87, in resource "proxmox_vm_qemu" "proxmox_vm":
│   87: resource "proxmox_vm_qemu" "proxmox_vm" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-proxmox_v2.9.14 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 41 [running]:
github.com/Telmate/proxmox-api-go/proxmox.NewConfigQemuFromApi(0xc0000c6718, 0xc9d509?)
	github.com/Telmate/proxmox-api-go@v0.0.0-20230319185744-e7cde7198cdf/proxmox/config_qemu.go:789 +0x4310
github.com/Telmate/terraform-provider-proxmox/proxmox.resourceVmQemuCreate(0xc000107880, {0xb66f60?, 0xc000264050})
	github.com/Telmate/terraform-provider-proxmox/proxmox/resource_vm_qemu.go:972 +0x2c4d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xdd7840?, {0xdd7840?, 0xc000445ec0?}, 0xd?, {0xb66f60?, 0xc000264050?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:695 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0002f67e0, {0xdd7840, 0xc000445ec0}, 0xc0001f7380, 0xc000107700, {0xb66f60, 0xc000264050})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00038eb70, {0xdd7840?, 0xc000445d10?}, 0xc0002866e0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.25.0/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000000460, {0xdd7840?, 0xc0004452c0?}, 0xc00010bc00)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xc6bc20?, 0xc000000460}, {0xdd7840, 0xc0004452c0}, 0xc00010bb90, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003a8000, {0xddb420, 0xc00023e1a0}, 0xc0002a8ea0, 0xc0003aacf0, 0x128f7a0, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1336 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc0003a8000, {0xddb420, 0xc00023e1a0}, 0xc0002a8ea0, 0x0)
	google.golang.org/grpc@v1.53.0/server.go:1704 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.53.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.53.0/server.go:963 +0x28a

Error: The terraform-provider-proxmox_v2.9.14 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

I'm using terraform version 1.5.3

This is what the .tf config file that I'm running:
```
terraform {
	required_providers {
		proxmox = {
			source = "Telmate/proxmox"
			version = "2.9.14"
		}
  	}
}

provider "proxmox" {
	pm_parallel = 1
	pm_tls_insecure = false
	pm_user = "root@pam"
	pm_log_enable = true
	pm_log_file = "terraform-plugin-proxmox.log"
	pm_log_levels = {
		 _default = "debug"
		 _capturelog = ""
	}
}

# In Terraform 0.12 and later, the templatefile function offers a built-in mechanism for rendering a template from a file. Use that instead of `template_file` data source.

# Source the cloud-init config file
data "template_cloudinit_config" "work" {
	for_each = var.instances
	# split in parts - 1st is cloud-init cfg as such; from 2nd onwards, shell scripts.
	# default gzip is true + base64 encoded (for proxmox don't encode or zip the cloud-init)
	gzip = false
	base64_encode = false
	part {
		filename = "cloud-init.cfg"
		content_type = "text/cloud-config"
		content = templatefile("${path.module}/files/cloud_init_ub2204.tpl", {
			root_ssh_key = "${each.value["ssh_key"]}"
			example-cert = file("${path.module}/files/example-cert.yaml")
			haproxy-deploy = file("${path.module}/files/haproxy-deploy.yaml")
			haproxy-ingress = file("${path.module}/files/haproxy-ingress.yaml")
			dockerconfigjson = file("${path.module}/files/dockerconfigjson")
  			tf_hostname = each.key
			dns_server = var.dns_server
			ip_address = each.value["ip_address"]
		})
	}
	part {
		filename = "kube-init.sh"
		content_type = "text/x-shellscript"
		content = templatefile("${path.module}/files/kube-init.sh", {
  			tf_hostname = each.key
			dns_server = var.dns_server
		})
	}
}

# Create a local copy of the template to transfer to Proxmox. `rendered` refers to interpreting the template.
resource "local_file" "cloud_init_ub2204" {
  	for_each = var.instances
	content = data.template_cloudinit_config.work["${each.key}"].rendered
	filename = "${path.module}/files/rendered-templates/cloud_init_ub2204-${each.key}.yaml"
	file_permission = 0640
}

# Transfer cloud-init to proxmox snippets (connect to the hypervisor)
resource "null_resource" "transfer_cloud_init" {
# trick using triggers and build_number to make the null_resource to execute each time and not store its state
	for_each = var.instances
	triggers = {
		build_number = "${timestamp()}"
	}
	connection {
		type = "ssh"
		user = "root"
		private_key = file("~/.ssh/id_ed25519")
		agent = false
		host = "pve1.example.com"
		port = 7800
	}
	provisioner "file" {
		# if you use template:
		# source = local_file.cloud_init_ub2204
		source = "files/rendered-templates/cloud_init_ub2204-${each.key}.yaml"
		destination = "/mnt/pve/cloudinit/snippets/cloud_init_ub2204-${each.key}.yaml"
	}
}

# Create the virtual machine
resource "proxmox_vm_qemu" "proxmox_vm" {
	for_each = var.instances

# cloud-init config needs to exist before continuing
	depends_on = [
		null_resource.transfer_cloud_init
	]
	name = "${each.key}-work"
	target_node = var.targetnode

# provision from clone:
	clone = "work.example.internal"
	full_clone = "true"
	os_type = "cloud-init"

# cloud-init options:
	cicustom = "user=cloudinit:snippets/cloud_init_ub2204-${each.key}.yaml"
	cloudinit_cdrom_storage = "cloudinit"
	ipconfig0 = "ip=${each.value["ip_address"]}/24,gw=${var.gw}"
	nameserver = "10.88.88.126"
	searchdomain = "example.com"

# assign resources and set general options:
	cores = "4"
	sockets = "1"
	cpu = "kvm64"
	memory = "12288"
	scsihw = "virtio-scsi-pci"
	bootdisk = "scsi0"
	boot = "c"
	force_create = "false"
	bios = "seabios"
	agent = 1
	numa = true
	hotplug = "network,disk,memory,cpu"
	balloon = "0"
	disk {
	  size = "100G"
	  type = "scsi"
	  storage = "local-lvm"
	  cache = "writeback"
	  ssd = "0"
	}
	network {
	  model = "virtio"
	  bridge = "vmbr1"
	}
	serial {
	  id = "0"
	  type = "socket"
	}

# Ignore changes to the network. MAC address is generated on every apply, causing
# TF to think this needs to be rebuilt on every apply
	lifecycle {
		ignore_changes = [
			network
	   	]
	}
}
```
I'm not sure how to debug it further, because it isn't saying exactly what is trying to convert into string.

**Comment by lethargosapatheia** [comment]
With the same configuration applied on Proxmox 7.4-3 I get:
```
╷
│ Error: scsi0 - cloud-init drive is already attached at 'ide0'
│
│   with proxmox_vm_qemu.proxmox_vm["matejamaric"],
│   on main.tf line 87, in resource "proxmox_vm_qemu" "proxmox_vm":
│   87: resource "proxmox_vm_qemu" "proxmox_vm" {
```
Which I can't understand, really. I wonder if the same thing happens in the background on Proxmox 6.4-15, but the API is not smart enough to report it.

**Comment by lethargosapatheia** [comment]
A little bit a correction:
```
-scsi0 local-lvm:vm-108-cloudinit
```
This is the parameter for scsi0 for some weird reason (the ID of the VM here is 108).

In a much older configuration (from more than a year ago), I have:
```
-scsi0 local-lvm:vm-121-disk-0
```
This is the correct one, because scsi0 is where the main disk resides. I don't understand how the terraform configuration translates into trying to attach the cloud-init drive to scsi0.

**Comment by lethargosapatheia** [comment]
I've just switched to version 2.9.11 which also has a similar error:
```
()
update VM 108: -cdrom cloudinit:cloudinit
TASK ERROR: ide2 - cloud-init drive is already attached at 'ide0'
```
The huge difference though is that it actually works. The error is ignored and it goes on to configure everything and the provisioning is finished.

I tried several version, but 2.9.12 wasn't available in the terraform registry (see https://github.com/Telmate/terraform-provider-proxmox/issues/815), so I ended up using 2.9.11 for now.
